nav-tracking-mitigations/stateful-client-bounce.sub.https.html

```diff
@@ -0,0 +1,50 @@
+<!DOCTYPE HTML>
+<meta charset=utf-8>
+<title>Bounce Tracking Mitigations: Stateful Client Bounce</title>
+<body>body for test_driver.bless</body>
+<script src=/resources/testharness.js></script>
+<script src=/resources/testharnessreport.js></script>
+<script src=/resources/testdriver.js></script>
+<script src=/resources/testdriver-vendor.js></script>
+<script>
+  promise_setup(async () => {
+    try {
+      await test_driver.set_storage_access('*', '*', 'blocked');
+    } catch (e) {
+      // Ignore, can be unimplemented if the platform blocks cross-site
+      // cookies by default. If this failed without default blocking the test
+      // will fail.
+    }
+  });
+
+  promise_test(async t => {
+    const {promise: getLinkPromise, resolve: getLinkResolver} =
+      Promise.withResolvers();
+
+    // Open a new tab and retrieve the link to start the test.
+    await test_driver.bless('open page to start test', async () => {
+      const otherTab = window.open(
+        'resources/redirect-chain-start.sub.https.html');
+
+      await new Promise(resolve => otherTab.addEventListener('load', resolve));
+      getLinkResolver(otherTab.document.getElementById('navigate-link'));
+    });
+
+    // Click the link in the new tab.
+    //
+    // Ideally, we would click this from within the page, but because the page
+    // immediately navigates away, test_driver fails as it expects the page to
+    // not navigate away. Doing this from the main test page avoids the issue.
+    const link = await getLinkPromise;
+    test_driver.click(link);
+
+    await new Promise(resolve => window.addEventListener('message', resolve));
+
+    let result = [];
+    while (result.length == 0) {
+      result = await test_driver.run_bounce_tracking_mitigations();
+    }
+
+    assert_array_equals(result, ['{{hosts[alt][]}}']);
+  }, 'Bounce tracking mitigations deleting state for a bounce tracker');
```

Comment:
We should also assert that the cookie has been deleted. This only asserts that the algorithm says it purged.